validator/src/services/mixmining/db.rs

```diff
@@ -0,0 +1,105 @@
+use super::Mixnode;
+
+/// A (currently RAM-based) data store to keep tabs on which nodes have what
+/// stake assigned to them.
+#[derive(Clone, Debug, PartialEq)]
+pub struct MixminingDb {
+    mixnodes: Vec<Mixnode>,
+    capacity: usize,
+}
+
+impl MixminingDb {
+    pub fn new() -> MixminingDb {
+        let mixnodes = Vec::<Mixnode>::new();
+        MixminingDb {
+            capacity: 6,
+            mixnodes,
+        }
+    }
+
+    pub fn add(&mut self, mixnode: Mixnode) {
+        self.mixnodes.push(mixnode);
+    }
+
+    pub fn get_mixnodes(&self) -> &Vec<Mixnode> {
```

Comment:
I'm not entirely sure whether returning `&Vec<_>` is the best idea here. I can't give a good reason against it, but it just feels weird for me. It might as well be totally fine thing to do that does everything we want, so also keep that in mind.